src/utils/mask.ts

```diff
@@ -0,0 +1,65 @@
+import { MaskableObject, MaskInput } from "../types";
+
+/**
+ * Recursively masks specified fields in an object or array with '***'.
+ * @param {MaskInput} obj - The object or array to mask. Can be undefined.
+ * @param {string[]} [fieldsToMask] - Array of field names to mask.
+ *                                    Defaults to ['createdBy', 'updatedBy', 'userName', 'userEmail', 'userRole']
+ * @returns {MaskInput} The object or array with specified fields masked
+ */
+const mask = (
+  obj: MaskInput,
+  fieldsToMask = [
+    "userName",
+    "userEmail",
+    "userRole",
```

Comment:
Can you please clarify why we're masking the userRole? It is important for debugging access control.

Reply:
No particular reason, I can remove it from here. I thought we only wanted id.

Reply:
Resolved this, only kept userName, userRole. Additional fields can be provided while initializing the logger